tests/op_test.py

```diff
@@ -0,0 +1,267 @@
+            else:
+                shape = list(np_value.shape)
+                lod_level = 0
+
+        persistable = True if name in persistable_list else False
+        return block.create_var(
+            dtype="float32",
+            shape=shape,
+            persistable=persistable,
+            lod_level=lod_level,
+            name=name)
+
+    # Go through all the variables in the expected list for this operator.
+    var_dict = {}
+    for var_proto in proto_list:
+        var_name = str(var_proto.name)
+
+        # If these are inputs, and the expected input is not necessary
+        # are is not provided in the list of inputs, we move on to the next
+        # expected. input.
+        # If not, we make sure it the expected input is provided, or that it
+        # is unnecessary.
+        if is_input:
+            if (var_name not in np_list) and var_proto.dispensable:
+                continue
+            assert (var_name in np_list) or (var_proto.dispensable), \
+                "Missing {} as input".format(var_name)
+
+        # Set duplicable variables as lists of Paddle variables, and standard
+        # ones as simple Paddle variables.
+        if var_proto.duplicable:
+            assert isinstance(np_list[var_name], list), \
+                "Duplicable {} should be set as list".format(var_name)
+            var_list = []
+            for (name, np_value) in np_list[var_name]:
+                var_list.append(
+                    create_var(block, name, {name: np_value}, var_proto))
+            var_dict[var_name] = var_list
+        else:
+            var_dict[var_name] = create_var(block, var_name, np_list, var_proto)
+
+    return var_dict
+
+
+class OpTest(unittest.TestCase):
+    """Evaluates an op maker's validity.
+
+    Using op-specific inputs and attributes, executes:
+    (1) A Paddle program
+    (2) A Caffe2 backend consuming Paddle ops converted to ONNX using custom
+        op makers.
+
+    It uses the outputs of both fo these executions to compare the values
+    of their outputs. Success in these comparisons comes from almost equal
+    values of this output data across both executions.
+
+    Attributes:
+        inputs (dict): Operator input name -> input value.
+        outputs (dict): Operator output -> output value placeholders.
+
+        Additionally, custom attributes to the op.
+    """
+    def feed_var(self, input_vars, place):
+        """Returns a dictionary of variable names -> initialized tensors.
+
+        It sets tensors' execution place set (CPU or GPU), and Level of
+        Details (LoD) using this info from the numpy values.
+        """
+        feed_map = {}
+        for var_name in input_vars:
+            if isinstance(input_vars[var_name], list):
+                for name, np_value in self.inputs[var_name]:
+                    tensor = core.LoDTensor()
+                    if isinstance(np_value, tuple):
+                        tensor.set(np_value[0], place)
```

Comment:
It would be nice to make new function for this code that repeats twice doing the `.set`s

Reply:
It is true. I will fix it in next pull request.